### DISCUSSION TITLE
advanced parameters documentation.

### DISCUSSION THREAD

**Discussion by arjunkrishna** [open-webui/open-webui#3794] [Q&A] [answered]
![image](https://github.com/open-webui/open-webui/assets/5271912/b380af68-ed46-4871-843e-12a7bface91a)

Is there a documentation on the https://docs.openwebui.com/ on how to use these parameters?

**Comment by clang88** [comment, marked as answer]
Some will be already documented here: 
https://platform.openai.com/docs/api-reference/chat/create
Others are documented on Ollama:
https://github.com/ollama/ollama/blob/main/docs/modelfile.md

**Comment by arjunkrishna** [comment]
thank you.

**Comment by Khampol** [comment]
Here’s a simple breakdown of each parameter with examples:

1. **Seed**: Ensures the same result every time. Example: With a set seed, asking for “Write a poem” will give the same poem every time.

2. **Stop Sequence**: Stops output at a specific word or phrase. Example: “End here” will stop the generation when the model reaches that phrase.

3. **Temperature**: Controls creativity. Example: Low temperature (0.2) makes answers precise; high temperature (0.9) makes them more random and varied.

4. **Mirostat**: Maintains a set creativity level. Example: Like a thermostat for creativity, it adjusts based on your target level.

5. **Mirostat Eta/Tau**: Fine-tune Mirostat’s sensitivity and stability. Example: Use these to make Mirostat adapt slower or faster to text generation.

6. **Top K**: Limits choices to the most likely words. Example: Setting Top K to 50 makes the model choose from only the top 50 words.

7. **Top P**: Adjusts creativity by probability. Example: Top P of 0.9 makes the model choose from words that together cover 90% probability.

8. **Frequency Penalty**: Reduces word repetition. Example: High penalty makes the model avoid reusing words like “awesome” too often.

9. **Repeat Last N**: Prevents recent phrase repeats. Example: Setting to 5 prevents repeating any words from the last 5 tokens.

10. **Tfs Z**: Similar to Top K but with frequency adjustments. Example: Limits the output to the most relevant words by their adjusted frequency.

11. **Context Length**: Defines how much text the model can remember. Example: A longer length helps it remember your earlier conversation better.

12. **Batch Size (num_batch)**: Sets how many responses to create. Example: If set to 3, it generates three different responses at once.

13. **Tokens to Keep on Context Refresh (num_keep)**: Retains part of the previous conversation. Example: Keeping 50 tokens helps the model remember the main topic when refreshing.

14. **Max Tokens (num_predict)**: Limits the output length. Example: Setting to 100 will make the response concise, capping it at 100 words.

(A little more.. ;) ...)

15. **use_mmap (Ollama)**: 
   - **Definition**: Enables or disables the use of memory-mapped files.
   - **Effect**: When enabled, the model can load parts of its weight file into memory on demand, which may improve performance by reducing RAM consumption.

16. **use_mlock (Ollama)**:
   - **Definition**: Enables or disables memory locking.
   - **Effect**: When enabled, this parameter prevents the memory used by the model from being swapped to disk. This can enhance performance by ensuring that the data remains in RAM.

17. **num_thread (Ollama)**:
   - **Definition**: Sets the number of threads to use during model execution.
   - **Effect**: More threads can improve performance by allowing the model to perform multiple tasks simultaneously, especially on multi-core systems.

18. **num_gpu (Ollama)**:
   - **Definition**: Sets the number of GPUs to use for running the model.
   - **Effect**: Using multiple GPUs can significantly accelerate processing, especially for large models or datasets.

These parameters are primarily used to optimize the model's performance based on available hardware resources and specific application needs.

**Reply by arjunkrishna** [reply]
Thank you.

**Reply by rgaricano** [reply]
13. **Tokens to Keep on Context Refresh (num_keep)**: Retains part of the previous conversation. It's used when the n_ctx limit is reached. A new prompt will be constructed with the first n_keep characters of the original prompt plus the second half of the output to free up space for more conversation. Example: Keeping 50 first tokens helps the model remember the main topic when refreshing.

**Comment by Consultleo** [comment]
Hi, is there anyway I can save these Advanced settings and apply them by default for all users?
Normally shouldnt these settings be somewhere in the model customization?

**Reply by silenceroom** [reply]
see the docs here:
https://docs.openwebui.com/features/chat-features/chat-params#system-prompt-and-advanced-parameters-hierarchy-chart

In the Models page from admin panel, you can change these items for all users.
